trailblazer/store/api.py

```diff
@@ -130,128 +205,198 @@ def user(self, email: str) -> models.User:
         """Fetch a user from the database."""
         return self.User.query.filter_by(email=email).first()
 
-    def aggregate_failed(self, since_when: dt.date = None) -> List:
-        """Count the number of failed jobs per category (name)."""
-
-        categories = self.session.query(
-            self.Job.name.label("name"),
-            sqa.func.count(self.Job.id).label("count"),
-        ).filter(self.Job.status != "cancelled")
-
-        if since_when:
-            categories = categories.filter(self.Job.started_at > since_when)
-
-        categories = categories.group_by(self.Job.name).all()
-
-        data = [{"name": category.name, "count": category.count} for category in categories]
-        return data
-
-    def jobs(self):
+    def jobs(self) -> Query:
         """Return all jobs in the database."""
         return self.Job.query
 
-    # CG code
-
-    def mark_analyses_deleted(self, case_id: str) -> None:
+    def mark_analyses_deleted(self, case_id: str) -> Query:
         """ mark analyses connected to a case as deleted """
-        for old_analysis in self.analyses(family=case_id):
-            old_analysis.is_deleted = True
+        old_analyses = self.analyses(case_id=case_id)
+        if old_analyses.count() > 0:
+            for old_analysis in old_analyses:
+                old_analysis.is_deleted = True
         self.commit()
+        return old_analyses
```

Comment:
Could you not loop even if count is 0? In that case skip this line. Otherwise I would prefer
```suggestion
        old_analyses = self.analyses(case_id=case_id)
        if old_analyses.count() == 0:
            return old_analyses
        for old_analysis in old_analyses:
            old_analysis.is_deleted = True
        self.commit()
        return old_analyses
```